EssentialFeed/EssentialFeedTests/ImageComments API/RemoteImageCommentLoaderTests.swift

```diff
@@ -0,0 +1,175 @@
+//
+//  RemoteImageCommentLoaderTests.swift
+//  EssentialFeedTests
+//
+//  Created by Eric Garlock on 2/28/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+import EssentialFeed
+
+class RemoteImageCommentLoaderTests: XCTestCase {
+	
+	func test_init_doesNotRequestDataFromURL() {
+		let (_, client) = makeSUT()
+		
+		XCTAssertEqual(client.requestedURLs, [])
+	}
+	
+	func test_load_requestsDataFromURL() {
+		let url = anyURL()
+		let (sut, client) = makeSUT(url: url)
+		
+		sut.load() { _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url])
+	}
+	
+	func test_load_requestsDataFromURLTwice() {
+		let url = anyURL()
+		let (sut, client) = makeSUT(url: url)
+		
+		sut.load() { _ in }
+		sut.load() { _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url, url])
+	}
+	
+	func test_load_deliversErrorOnClientError() {
+		let (sut, client) = makeSUT()
+		
+		expect(sut, toCompleteWith: .failure(RemoteImageCommentLoader.Error.connectivity)) {
+			client.complete(with: anyNSError())
+		}
+	}
+	
+	func test_load_deliversErrorOnNon200HTTPResponse() {
+		let (sut, client) = makeSUT()
+		
+		let codes = [199, 201, 300, 400, 500]
+		
+		codes.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: .failure(RemoteImageCommentLoader.Error.invalidData)) {
+				client.complete(withStatusCode: code, data: anyData(), at: index)
```

Comment:
`anyData` is not a valid response for this endpoint based on API the contract. It should use a valid response to be able to test the `status code` logic.

It could be an empty JSON list: 

```suggestion
				let emptyListJSONData = "{\"items\": []}".data(using: .utf8)!
				client.complete(withStatusCode: code, data: emptyListJSONData, at: index)
```